code/game/objects/items/stacks/stack.dm

```diff
@@ -355,7 +357,11 @@
 	use(amount)
 
 /obj/item/stack/attackby(obj/item/W, mob/user, params)
-	if(istype(W, merge_type))
+	if(!parent_stack && istype(W, merge_type))
+		var/obj/item/stack/S = W
+		merge(S)
+		to_chat(user, "<span class='notice'>Your [S.name] stack now contains [S.get_amount()] [S.singular_name]\s.</span>")
+	if(parent_stack && W.type == type)
 		var/obj/item/stack/S = W
 		merge(S)
 		to_chat(user, "<span class='notice'>Your [S.name] stack now contains [S.get_amount()] [S.singular_name]\s.</span>")
```

Comment:
Since these have the same outcomes, they can be simplified to 
```suggestion
	if(!parent_stack && istype(W, merge_type) || parent_stack && W.type == type)
		var/obj/item/stack/S = W
		merge(S)
		to_chat(user, "<span class='notice'>Your [S.name] stack now contains [S.get_amount()] [S.singular_name]\s.</span>")
```

Reply:
This is much cleaner! updated to this in latest commit.